src/test/compile-fail/no-core-gen-trait.rs

```diff
@@ -0,0 +1,20 @@
+// error-pattern: requires `generator` lang_item
```

Comment:
Move this to `src/test/ui` (preferably colocated with some other lang item tests). Nothing new should be added to `compile-fail` unless you're having specific CI problems.